### PR TITLE
INT-B-20492

### PR DESCRIPTION
**Pull request by loganwc**
## Summary

Added bulk download to Payment Request Document Viewer. Also added ability to convert images to pdf _without_ rotating the images to generator.go

### How to test

1. Access the office app
2. Login as a TIO
3. Open a move with Proof of Service docs (ORGSIT from seed data)
4. Review Payment Request and approve
5. Click View Documents
6. Click `Download All Files (PDF)` button at top of document viewer
7. Accept download and review pdf has all proof of service uploads

If you want to add proof of service documentation on your own move login as Prime and click on your move. 
At the bottom of the page there are options to `Upload Document for ....` you can try uploading documents there (it doesn't always work) or you can try Uploading Document after you created your Payment Request at the bottom of the page.

> [!NOTE]  
> Wait a couple seconds after it says `Upload Complete` 

Screenshot 2024-09-05 at 1 00 06 PM

### CODE REVIEW

**Review by pambecker — APPROVED** [review]

**Comment by paulstonebraker** [comment]
image
Can we find a different spot for the button? It adds a vertical scroll to the whole page

**Comment by paulstonebraker** [comment]
> image Can we find a different spot for the 'Download All Files' button? It adds a vertical scroll to the whole page

suggestion: supporting documents and order have a 'Download File' button here... since we don't have that for the TIO view documents, maybe put it here? otherwise maybe in the hamburger button?

image

**Comment by loganwc** [comment]
> > image Can we find a different spot for the 'Download All Files' button? It adds a vertical scroll to the whole page
> 
> suggestion: supporting documents and order have a 'Download File' button here... since we don't have that for the TIO view documents, maybe put it here? otherwise maybe in the hamburger button?
> 
> image

That would require me to modify the document viewer and prop drill all that information and it would've been pretty messy

File: swagger-def/ghc.yaml [hunk 1]
```diff
@@ -3141,6 +3141,40 @@ paths:
...
+              type: string
+              description: File name to download
+          description: Payment Request Files PDF
+          schema:
+            format: binary
+            type: file
+        '400':
+          $ref: '#/responses/InvalidRequest'
+        '403':
+          $ref: '#/responses/PermissionDenied'
```
Comment (paulstonebraker) [inline swagger-def/ghc.yaml:3173]: I don't believe 403 or 404 are used, if so remove them pleasies

**Review by paulstonebraker — CHANGES_REQUESTED** [review]

**Comment by paulstonebraker** [comment]
> > > image Can we find a different spot for the 'Download All Files' button? It adds a vertical scroll to the whole page
> > 
> > 
> > suggestion: supporting documents and order have a 'Download File' button here... since we don't have that for the TIO view documents, maybe put it here? otherwise maybe in the hamburger button?
> > image
> 
> That would require me to modify the document viewer and prop drill all that information and it would've been pretty messy

okay, where else can we move it so its not there 😄

**Comment by paulstonebraker** [comment]
Very nicely done, code works. I think `Download all files` button should be moved but if team over rules me I will happily approve.

**Comment by pambecker** [comment]
> Very nicely done, code works. I think `Download all files` button should be moved but if team over rules me I will happily approve.

Paul, if you move off of this branch and pull up the View Docs, that scroll bar is not there?  I wasn't seeing a scroll bar when I tested, but my doc was not full page I guess?

**Comment by paulstonebraker** [comment]
> > > Very nicely done, code works. I think `Download all files` button should be moved but if team over rules me I will happily approve.
> > 
> > 
> > Paul, if you move off of this branch and pull up the View Docs, that scroll bar is not there? I wasn't seeing a scroll bar when I tested, but my doc was not full page I guess?
> 
> I'm not sure, I will check. Personally, I just don't like the `Download all files` link taking up that entire section of my screen. But if it doesn't bug anyone else besides me, I certainly won't let it block my approval 😺

Just checked and viewing the same move/document, scroll did not appear off this branch.

**Comment by pambecker** [comment]
> > > > Very nicely done, code works. I think `Download all files` button should be moved but if team over rules me I will happily approve.
> > > 
> > > 
> > > Paul, if you move off of this branch and pull up the View Docs, that scroll bar is not there? I wasn't seeing a scroll bar when I tested, but my doc was not full page I guess?
> > 
> > 
> > I'm not sure, I will check. Personally, I just don't like the `Download all files` link taking up that entire section of my screen. But if it doesn't bug anyone else besides me, I certainly won't let it block my approval 😺
> 
> Just checked and viewing the same move/document, scroll did not appear off this branch.

yeah, I agree, we don't want to introduce a scroll if there's a way around it.  Logan can you see about a way to prevent the scroll?

File: swagger-def/ghc.yaml (hunk 1 above)
Comment (paulstonebraker) [reply swagger-def/ghc.yaml:3173]: 😬

**Pull request merged**